parsing/location.ml

```diff
@@ -459,20 +459,20 @@ let highlight_quote ppf
         (* Single-line error *)
         Format.fprintf ppf "%s | %s@," line_nb line;
         Format.fprintf ppf "%*s   " (String.length line_nb) "";
-        String.iteri (fun i c ->
+        for i = 0 to rightmost.pos_cnum do
```

Comment:
Agree it's complicated ! Added a comment.

I think the code above works because the end of line has an indice (there's a `pos_cnum` that correspond to it), so only one line can be picked if the location spans it (it won't pick the start of the next line).
I ended up rewriting it in https://github.com/ocaml/ocaml/pull/11678 (with more comments)

Reply:
Wait, `rightmost.pos_cnum` counts the number of character since the start of the file. It is far too large, the right size is `rightmost.pos_cnum - line_cnum_start - 1`.

Reply:
Right! That was a big oversight, thanks.